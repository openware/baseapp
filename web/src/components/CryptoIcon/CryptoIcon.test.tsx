import { shallow } from 'enzyme';
import { TestComponentWrapper } from 'lib/test';
import * as React from 'react';
import { CryptoIcon, CryptoIconProps } from '.';

const defaultProps: CryptoIconProps = {
    code: 'btc',
};

const setup = (props: Partial<CryptoIconProps> = {}) =>
    shallow(
        <TestComponentWrapper>
            <CryptoIcon {...{ ...defaultProps, ...props }} />
        </TestComponentWrapper>
    );

describe('CryptoIcon', () => {
    it('should render', () => {
        const wrapper = setup().render();
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        const wrapper = setup().render();
        expect(wrapper.hasClass('cr-crypto-icon')).toBeTruthy();
    });

    it('should render children', () => {
        const amount = 123;
        const wrapper = setup({ children: [amount] }).render();
        expect(wrapper.text()).toEqual(` ${amount}`);
    });

    it.skip('should have correct path to svg images', () => {
        const wrapper = setup().render();
        expect(wrapper.find('img').prop('src')).toEqual(require('cryptocurrency-icons/svg/color/btc.svg'));
    });

    it('should pass along supplied className', () => {
        const className = 'some-class';
        const wrapper = setup({ className }).render();
        expect(wrapper.hasClass(className)).toBeTruthy();
    });
});
