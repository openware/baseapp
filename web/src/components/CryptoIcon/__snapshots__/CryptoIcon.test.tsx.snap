// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CryptoIcon should render 1`] = `
<span
  class="cr-crypto-icon"
>
  <img
    alt="btc"
  />
   
</span>
`;
